src/main/java/database/service/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @Override
+    public void removeAllRecordsFromTable(Class<? extends BaseEntity> entityClass) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Files.writeString(databasePath, "[]");
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T getById(Class<? extends BaseEntity> entityClass, I id) throws IOException,
+            ProvidedIdDoesNotExistException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        return entities.stream()
+                .filter(e -> id.equals(e.getId()))
+                .findFirst()
+                .orElseThrow(() -> new ProvidedIdDoesNotExistException("Entity with provided Id does not exist."));
+    }
+
+    @Override
+    public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass)
+            throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        return deserializeEntities(entityClass, Files.readString(databasePath));
+    }
+
+    @Override
+    public <T extends BaseEntity, V> Iterable<T> getByFilters(Class<? extends BaseEntity> entityClass,
+                                                              Set<Pair<String, V>> filters) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        return entities.stream()
+                .filter(entity -> {
+                    for (Pair<String, V> filter : filters) {
+                        String propertyName = filter.getKey();
+                        V expectedValue = filter.getValue();
+
+                        try {
+                            Field field = entityClass.getDeclaredField(propertyName);
+                            field.setAccessible(true);
+                            Object actualValue = field.get(entity);
+
+                            if (!expectedValue.equals(actualValue)) {
+                                return false;
+                            }
+                        } catch (NoSuchFieldException | IllegalAccessException e) {
+                            e.printStackTrace();
+                            return false;
+                        }
+                    }
+                    return true;
+                })
+                .collect(Collectors.toList());
+    }
+
+    String getDatabasePath(Class<? extends BaseEntity> entityClass) {
+        var databaseFolder = settings.getDatabasePath();
+        return databaseFolder + File.separator + entityClass.getSimpleName() + "Table" + ".json";
+    }
+
+    private <T extends BaseEntity> void updateEntityFields(T outcomeEntity, T incomeEntity) {
+        Field[] fields = incomeEntity.getClass().getDeclaredFields();
+        for (Field field : fields) {
```

Comment:
We probably don't want to overwrite `id` field, it should be excluded from the list of fields.